source/language/types.rst

```diff
@@ -267,6 +267,38 @@ are all of type ``float[64]``.
 
 Note that `e` is a valid identifier. `e/E` are also used in scientific notation where appropriate.
 
+Mathematical functions available for constant initialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to simple arithmetic functions used in expressions initializing constants,
+OpenQASM 3 offers the following built-in mathematical operators followed by
+their argument expression in parentheses:
+
+.. container::
+   :name: tab:built-in-math
+
+   .. table:: Built-in mathematical functions in OpenQASM3.
+
+      +--------+
+      | arccos |
+      +--------+
+      | arcsin |
+      +--------+
+      | arctan |
+      +--------+
+      | cos    |
+      +--------+
+      | exp    |
+      +--------+
+      | ln     |
+      +--------+
+      | sin    |
+      +--------+
+      | sqrt   |
+      +--------+
+      | tan    |
+      +--------+
```

Comment:
We're missing `pow`, `floor`, `ceiling`, `div`, `mod` and `log` that are listed in the above paragraph at the moment.  Some things we'll need to mention:

- `pow`: for fractional powers, we need to say that it returns the "principal root", and specify how many arguments it takes (usually two, but sometimes a third, which is a modulus - I think we should only have the two-argument form).  It's questionable what input and output types this function should allow, given that it only has one name but `pow(-1.0, 0.5)` would be complex.  Perhaps for now it's best to say it's got overloads such that it allows
  - `pow(int, unsigned int) -> int`
  - `pow(float, int or float) -> float`
  - `pow(complex, int or float or complex) -> complex`
  
  and anything else is an error.

- `floor`, `ceiling`: these only make sense if they take a `float` argument.  Conceivably, they could return either a `float` or an `int`.  I think it's safest to return a `float`, and require casting to `int`; if you can represent the number `a.b` for digits `a` and `b` at a given floating-point precision, it's guaranteed that you can also exactly represent the integers `a` and `a + 1` in the same floating-point precision, so there's no risk there.

- `div`: not sure what was intended by this function when it was originally written into the spec.  In C it returns a struct of the quotient and remainder for efficiency reasons, but there's no need for that at compile time.  Possibly we should remove this from the list, but we can solicit suggestions.

- `mod`: assuming we mean modular division, we should have overloads for
  - `mod(int, int) -> int`
  - `mod(float, int or float) -> float`

  and we need to be sure to define what happens when any combination of the two arguments are negative.  Modular division doesn't seem (to me) to make sense on complex numbers (though we likely do want to include an `abs` function, which calculates the complex modulus).